### PR TITLE
Mostly image sizing and image lazy loading stuff. A few minor style t…

### PR DESCRIPTION
**Pull request by BjornTwachtmann**
…hings.

Please tell me if I'm being stupid but it looks like these were missed. Map images not being lazy loaded, sizes, etc.

### CODE REVIEW

File: src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php [hunk 1]
```diff
@@ -47,6 +47,19 @@ public function __construct(ProgrammeContainer $programmeContainer, ?Episode $st
         }
     }
 
+    public function getImageSizes()
```
Comment (votemike) [inline src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php]: Is this not also the same in the TxPresenter?

File: src/DsAmen/Organism/Map/MapPresenter.php [hunk 2]
```diff
@@ -247,13 +245,14 @@ private function showThirdColumn(bool $hasComingSoon)
         return $this->upcomingBroadcasts || $hasBroadcastInLast18Months || $hasComingSoon || $this->isWorldNews();
     }
 
-    private function shouldShowMiniMap(): bool
+    private function setShowMiniMap(): void
```
Comment (felipeparaujo) [inline src/DsAmen/Organism/Map/MapPresenter.php]: This doesn't seem to be called anywhere. Anyway, this should probably be done in the constructor

Comment (BjornTwachtmann) [reply src/DsAmen/Organism/Map/MapPresenter.php]: shit, I was calling that from the construtor, I think I lost that line somewhere, oops

Comment (felipeparaujo) [reply src/DsAmen/Organism/Map/MapPresenter.php]: have you tried looking behind the sofa? most of the lines i lose end up there

File: src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php (hunk 1 above)
Comment (BjornTwachtmann) [reply src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php]: Yeah, except without the full width option as far as I can see.

File: src/DsAmen/Organism/Map/MapPresenter.php (hunk 2 above)
Comment (BjornTwachtmann) [reply src/DsAmen/Organism/Map/MapPresenter.php]: It was wedged underneath the spacebar, along with some long-lost navel lint.

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: app/Resources/views/find_by_pid/tlec.html.twig [hunk 3]
```diff
@@ -31,10 +31,22 @@
             {%- if promotions or clips or galleries -%}
                 <ul class="grid">
                     {%- for promotion in promotions -%}
-                        {% set promo_widths = loop.first and has_double_width_first_promo ? '2/3@gel3b 1/2@gel5' : '1/2@gel3 1/3@gel3b 1/4@gel5' %}
+                        {# @TODO should this be moved out into a page presenter or something. It's fugly #}
```
Comment (votemike) [inline app/Resources/views/find_by_pid/tlec.html.twig]: Can that be done now?

Comment (BjornTwachtmann) [reply app/Resources/views/find_by_pid/tlec.html.twig]: Can be. I'm tempted to wait to see how much more logic ends up in this template before doing it. It's probably not worth it for this one thing.

File: app/Resources/views/find_by_pid/tlec.html.twig [hunk 4]
```diff
@@ -51,10 +63,18 @@
                                 'branding_name': 'secondary',
                                 'title_options': {
                                     'text_colour_on_title_link': true,
-                                    'title_size_small': 'gel-long-primer'
+                                    'title_size_small': 'gel-long-primer',
+                                },
+                                'image_options': {
+                                    'sizes': {
```
Comment (votemike) [inline app/Resources/views/find_by_pid/tlec.html.twig]: Are these sizes not dynamic like the promotions?

File: src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php (hunk 1 above)
Comment (votemike) [inline src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php]: Typehints?

File: app/Resources/views/find_by_pid/tlec.html.twig (hunk 4 above)
Comment (BjornTwachtmann) [reply app/Resources/views/find_by_pid/tlec.html.twig]: nein, there's no "double width first clip", thankfully

File: src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php (hunk 1 above)
Comment (BjornTwachtmann) [reply src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php]: ah, yep

File: src/Controller/FindByPid/TlecController.php [hunk 5]
```diff
@@ -75,24 +73,31 @@ public function __invoke(
...
-            'has_double_width_first_promo' => $hasDoubleWidthFirstPromo,
             'mapPresenter' => $mapPresenter,
         ]);
     }
 
     private function getComingSoonPromotion(ImagesService $imagesService, ProgrammeContainer $programme): ?Promotion
     {
         $comingSoonBlock = $programme->getOption('comingsoon_block');
-        if (empty($comingSoonBlock) || empty($comingSoonBlock['content']) || empty($comingSoonBlock['content']['promotions'])) {
+        if (empty($comingSoonBlock['content']['promotions'])) {
```
Comment (votemike) [inline src/Controller/FindByPid/TlecController.php]: 👍

**Comment by votemike** [comment]
LGTM

**Pull request merged**